src/Orleans.Streaming.Redis/Storage/RedisDataManager.cs

```diff
@@ -8,50 +8,83 @@
 using System.Linq;
 using Orleans.Providers.Streams.Redis;
 using Orleans.Configuration;
+using Serilog.Core;
+using Orleans.Redis.Common;
+using System.Threading;
 
 namespace Orleans.Streaming.Redis.Storage
 {
-    internal class RedisDataManager
+    internal class RedisDataManager : IRedisDataManager
     {
+        private const int MINIMUM_KEY_LENGTH = 5;
+        private const int MAXIMUM_KEY_LENGTH = 256;
+
         public const int UnlimitedMessageCount = -1;
 
         public string QueueName { get; private set; }
 
         private readonly RedisStreamOptions _options;
+        private readonly IConnectionMultiplexerFactory _connectionMultiplexerFactory;
         private readonly ILogger _logger;
 
         private readonly ConcurrentQueue<RedisValue> _queue = new ConcurrentQueue<RedisValue>();
         private readonly RedisChannel _redisChannel;
 
         private static readonly IEnumerable<RedisValue> EmptyRedisValueEnumerable = Array.Empty<RedisValue>();
 
+        private bool _initialized = false;
         private IConnectionMultiplexer _connectionMultiplexer;
 
-        private RedisDataManager(RedisStreamOptions options, ILogger loggerFactory, string queueName)
+        internal ConcurrentQueue<RedisValue> TestHook_Queue => _queue;
+
+        private RedisDataManager(RedisStreamOptions options, IConnectionMultiplexerFactory connectionMultiplexerFactory, ILogger loggerFactory, string queueName)
         {
             queueName = SanitizeQueueName(queueName);
             ValidateQueueName(queueName);
 
             _options = options;
-            _logger = loggerFactory.ForContext<RedisDataManager>();
+            _connectionMultiplexerFactory = connectionMultiplexerFactory;
+            _logger = (loggerFactory ?? SilentLogger.Logger).ForContext<RedisDataManager>();
```

Comment:
Ah I think I might like this pattern best for using the `SilentLogger` if a non-null `logger` isn't provided.